test/DynamoCoreWpfTests/NodeViewCustomizationTests.cs

```diff
@@ -393,5 +396,27 @@ public void WatchConnectDisconnectTest()
             DispatcherUtil.DoEvents();
             Assert.AreEqual(8, items.Count());
         }
+
+        [Test]
+        public void TestEditReadOnlyCustomNodeProperty()
+        {
+            // Open a read-only custom node
+            var pathInTestsDir = @"core\CustomNodes\add_Read_only.dyf";
+            var filePath = Path.Combine(GetTestDirectory(ExecutingDirectory), pathInTestsDir);
+            FileInfo fInfo = new FileInfo(filePath);
+            fInfo.IsReadOnly = true;
+            Assert.IsTrue(DynamoUtilities.PathHelper.IsReadOnlyPath(filePath));
+
+            // a file with a read-only custom node definition is opened
+            Open(@"core\CustomNodes\TestAdd.dyn");
+            var homeWorkspace = Model.CurrentWorkspace as HomeWorkspaceModel;
+            Assert.NotNull(Model.CurrentWorkspace);
+
+            var funcNode = homeWorkspace.Nodes.OfType<Function>().First();
+            var customNodeView = NodeViewWithGuid("fb872c7c-21af-4074-8011-818874738dc7");
+            
+            MenuItem menuItem = customNodeView.MainContextMenu.Items.GetItemAt(customNodeView.MainContextMenu.Items.Count - 2) as MenuItem;
```

Comment:
this indexing into the menu is sure to break in the future, can you use the resource string or name of the framework element to look it up?